### PR TITLE
ci: Add support for HTTPX 0.28

### PR DESCRIPTION
**Pull request by carl-baillargeon**
# Description

Need to install a fork of respx until https://github.com/lundberg/respx/issues/277 is fixed.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -29,7 +29,7 @@ dependencies = [
   "PyYAML>=6.0",
   "requests>=2.31.0",
   "rich>=13.5.2,<14",
-  "httpx>=0.27.0"
+  "httpx>=0.28.0"
```
Comment (gmuloc) [inline pyproject.toml:32]: I don't think we need to bump here right?

Comment (carl-baillargeon) [reply pyproject.toml:32]: True

Comment (carl-baillargeon) [inline pyproject.toml:32]:
```suggestion
  "httpx>=0.27.0"
```

File: pyproject.toml [hunk 2]
```diff
@@ -75,7 +75,7 @@ dev = [
   "pytest-httpx>=0.30.0",
   "pytest-metadata>=3.0.0",
   "pytest>=7.4.0",
-  "respx>=0.21.1",
+  "respx @ git+https://github.com/ndhansen/respx",
```
Comment (gmuloc) [inline pyproject.toml:78]:
```suggestion
  "respx @ git+https://github.com/ndhansen/respx",  # until https://github.com/lundberg/respx/pull/278 is merged
```

**Review by gmuloc — APPROVED** [review]

**Pull request merged**